### PR TITLE
fix(activatable): add listeners for custom activators

### LINKED ISSUE #5860: [Bug Report] external menu activator is not working in `open-on-hover` mode
### Versions and Environment
**Vuetify:** 1.3.12
**Vue:** 2.5.20
**Browsers:** Chrome 70.0.3538.110
**OS:** Windows 10

### Steps to reproduce
Hover the red button

### Expected Behavior
Menu should open

### Actual Behavior
Menu doesn't open

### Reproduction Link
<a href="https://codepen.io/anon/pen/xmbzGj?editors=1010" target="_blank">https://codepen.io/anon/pen/xmbzGj?editors=1010</a>

### Other comments
Works without `open-on-hover` prop (green button)

### LINKED ISSUE #5890: [Bug Report] v-tooltip and v-menu do not work with an external activator
### Versions and Environment
**Vuetify:** 1.3.14
**Vue:** 2.5.21
**Browsers:** Chrome 70.0.3538.110
**OS:** Mac OS 10.13.6

### Steps to reproduce
Create `v-tooltip` component, create a **separate** activator button.

### Expected Behavior
Tooltip should appear when hovering over activator

### Actual Behavior
Tooltip does not appear

### Reproduction Link
<a href="https://codepen.io/anon/pen/roxwoZ?&editable=true&editors=101" target="_blank">https://codepen.io/anon/pen/roxwoZ?&editable=true&editors=101</a>

### LINKED ISSUE #7944: [Bug Report] activatable mixin 'activator' prop does nothing
### Versions and Environment
**Vuetify:** 2.0.0
**Vue:** 2.6.10
**Browsers:** Chrome 75.0.3770.142
**OS:** Mac OS 10.14.5

### Steps to reproduce
Reproduce case is simply a fork of the reproduce case from #5860, updated to Vuetify 2.0. That used to work for clicking, just not hovering, now it works for neither.

### Expected Behavior
The `activator` prop should be usable. This means event listeners should be bound dynamically, and unbound if the activator is changed.

### Actual Behavior
It's not useable.

### Reproduction Link
<a href="https://codepen.io/anon/pen/voGNog" target="_blank">https://codepen.io/anon/pen/voGNog</a>

### Other comments
It's clear to see from the code (https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/mixins/activatable/index.ts) that this is simply not hooked up. There's no test coverage as well.

Despite 2 months ago me providing #7185 which fixed #5860, and commenting that this `activator` prop appeared hard broken in the `next` branch at the time, this still made it to 2.0 being broken.

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Add listeners to custom activators

fixes #7944
fixes #5890 
fixes #5860

## Motivation and Context
Custom activators were not working at all

## How Has This Been Tested?
jest

## Markup:

<details>

```vue
<template>
  <div class="ma-5 pa-5">
    <v-container>
      <v-btn
        id="activator-click"
        ref="act"
        color="success"
      >
        click
      </v-btn>

      <v-menu
        :activator="activator"
      >
        <v-card>
          <v-card-text>
          </v-card-text>
        </v-card>
      </v-menu>
      <div class="ma-5"></div>
      <v-btn
        id="activator-hover"
        color="error"
      >
        hover
      </v-btn>
      <v-menu
        activator="#activator-hover"
        open-on-hover
      >
        <v-card>
          <v-card-text>
          </v-card-text>
        </v-card>
      </v-menu>
    </v-container>
  </div>
</template>

<script>
  export default {
    data: () => ({
      activator: null,
    }),

    mounted () {
      this.activator = this.$refs.act
    },
  }
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/mixins/activatable/index.ts [hunk 1]
```diff
@@ -119,18 +127,53 @@ export default baseMixins.extend({
       if (this.activator) {
         const target = this.internalActivator ? this.$el : document
 
-        activator = typeof this.activator === 'string'
-          ? target.querySelector(this.activator)
-          : this.activator
+        // #foobar
```
Comment (sh7dm) [inline packages/vuetify/src/mixins/activatable/index.ts]:
```suggestion
        // Selector
```

File: packages/vuetify/src/mixins/activatable/index.ts [hunk 2]
```diff
@@ -50,18 +54,27 @@ export default baseMixins.extend({
...
+        !this.activator ||
+        this.disabled ||
+        !this.activatorElement
+      ) return
+
+      const listeners = this.genActivatorListeners()
+      const keys = Object.keys(listeners)
+
+      for (const key of keys) {
+        (this.activatorElement as any)[`on${key}`] = listeners[key]
```
Comment (KaelWD) [inline packages/vuetify/src/mixins/activatable/index.ts]: addEventListener?

File: packages/vuetify/src/mixins/activatable/index.ts [hunk 3]
```diff
@@ -119,18 +127,53 @@ export default baseMixins.extend({
...
+      }
+    },
+    removeActivatorEvents () {
+      if (
+        !this.activator ||
+        !this.activatorElement
+      ) return
+
+      for (const event of this.events) {
+        (this.activatorElement as any)[`on${event}`] = null
```
Comment (KaelWD) [inline packages/vuetify/src/mixins/activatable/index.ts]: removeEventListener?

File: packages/vuetify/src/mixins/activatable/index.ts (hunk 2 above)
Comment (johnleider) [reply packages/vuetify/src/mixins/activatable/index.ts]: There was a specific reason for this, I'll double check as to why.

File: packages/vuetify/src/mixins/activatable/index.ts (hunk 3 above)
Comment (johnleider) [reply packages/vuetify/src/mixins/activatable/index.ts]: There was a specific reason for this, I'll double check as to why.

**Pull request merged**